src/types.d.ts

```diff
@@ -0,0 +1,1194 @@
+  }
+
+  export interface HotspotOptions {
+    perspective?: Perspective
+  }
+
+  /**
+   * HotspotContainer
+   */
+  export class HotpsotContainer {
+    constructor(
+      parentDomElement: HTMLElement,
+      stage: Stage,
+      view: View,
+      renderLoop: RenderLoop,
+      opts?: { rect?: RectSpec }
+    )
+
+    createHotspot(
+      domElement: HTMLElement,
+      coords?: RectilinearCoords,
+      opts?: HotspotOptions
+    ): Hotspot
+
+    destroy(): void
+
+    destroyHotspot(hotspot: Hotspot): void
+
+    domElement(): HTMLElement
+
+    hasHotspot(hotspot: Hotspot): boolean
+
+    hide(): void
+
+    listHotspots(): Hotspot[]
+
+    rect(): Rect
+
+    setRect(rect: Rect): void
+
+    show(): void
+  }
+
+  /**
+   * HtmlImageLoader
+   */
+  export class HtmlImageLoader {
+    constructor(stage: Stage)
+
+    loadImage(url: string, rect?: Rect, done?: Function)
+  }
+
+  export interface HtmlImageLoaderInteface extends HtmlImageLoader {}
+
+  /**
+   * ImageUrlSource
+   */
+  export class ImageUrlSource implements Source {
+    constructor(
+      sourceFromTile: (tile: Tile) => { url: string; rect?: Rect },
+      opts?: { concurrency?: number; retryDelay?: number }
+    )
+
+    static fromString: (
+      url: string,
+      options?: {
+        cubeMapPreviewFaceOrder?: string
+        cubeMapPreviewUrl?: string
+        retryDelay?: number
+      }
+    ) => ImageUrlSource
+
+    loadAsset(stage: Stage, tile: Tile, done: Function): Function
+
+    addEventListener(event: Event, callback: Function)
+
+    removeEventListener(event: Event, callback: Function)
+  }
+
+  /**
+   * KeyControlMethod
+   */
+  export class KeyControlMethod {
+    constructor(
+      keyCode: number,
+      parameter: string,
+      velocity: number,
+      friction: number,
+      element?: HTMLElement
+    )
+
+    destroy(): void
+  }
+
+  /**
+   * Layer
+   */
+  export class Layer {
+    constructor(
+      source: Source,
+      geometry: Geometry,
+      view: View,
+      textureStore: TextureStore,
+      opts?: LayerOptions
+    )
+
+    destroy(): void
+
+    effects(): Effects
+
+    fixedLevel(): number | null
+
+    geometry(): Geometry
+
+    mergeEffects(effects: Effects): void
+
+    pinFirstLevel(): void
+
+    setEffects(effects: Effects): void
+
+    setFixedLevel(levelIndex: number | null): void
+
+    source(): Source
+
+    textureStore(): TextureStore
+
+    unpinFirstLevel(): void
+
+    unpinLevel(levelIndex: number | null): void
+
+    view(): View
+
+    addEventListener(event: Event, callback: Function)
+
+    removeEventListener(event: Event, callback: Function)
+  }
+
+  export interface LayerInteface extends Layer {}
+
+  interface LayerOptions {
+    effects?: Effects
+  }
+
+  /**
+   * NetworkError
+   */
+  export class NetworkError {}
+
+  /**
+   * PinchZoomControlMethod
+   */
+  export class PinchZoomControlMethod {
+    constructor(element: HTMLElement, pointerType: string, opts: any)
```

Comment:
The [docs](https://www.marzipano.net/reference/PinchZoomControlMethod.html) are not helpful regarding `opts`.

According to the [code](https://github.com/google/marzipano/blob/eb116412d1b7c6df08906e9affcc9d3d65b5906f/src/controls/PinchZoom.js#L34), this parameter is not even in use.

Maybe we should not expose it in the type definitions.